apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,674 @@
+          });
+
+          // make the file inputs display the file name
+          $(document).on('change', '.custom-file-input', function () {
+            this.fileName = $(this).val().replace(/\\\\/g, '/').replace(/.*\\//, '');
+            $(this).parent('.custom-file').find('.custom-file-label').text(this.fileName);
+          });
+        </script>
+        ${renderEjs(__filename, "<%- include('../partials/navbar'); %>", resLocals)}
+        <main id="content" class="container-fluid">
+          ${renderEjs(
+            __filename,
+            "<%- include('../partials/assessmentSyncErrorsAndWarnings'); %>",
+            resLocals,
+          )}
+          ${typeof resLocals.errormsg !== 'undefined' && resLocals.errormsg.length > 0
+            ? html` <div class="alert alert-danger" role="alert">${resLocals.errormsg}</div> `
+            : ''}
+          ${!resLocals.isGroup
+            ? html`
+                <div class="card mb-4">
+                  <div class="card-header bg-primary text-white d-flex align-items-center">
+                    ${resLocals.assessment_set.name} ${resLocals.assessment.number}: Groups
+                  </div>
+                  <div class="card-body">
+                    This is not a group assessment. To enable this functionality, please set
+                    <code>"groupWork": true</code> in <code>infoAssessment.json</code>.
+                  </div>
+                </div>
+              `
+            : html`
+                ${resLocals.authz_data.has_course_instance_permission_edit
+                  ? html`
+                      <div class="container-fluid">
+                        <div
+                          class="modal fade"
+                          id="uploadAssessmentGroupsModal"
+                          tabindex="-1"
+                          role="dialog"
+                          aria-labelledby="uploadAssessmentGroupsModalLabel"
+                        >
+                          <div class="modal-dialog" role="document">
+                            <div class="modal-content">
+                              <div class="modal-header">
+                                <h1 class="h4 modal-title" id="uploadAssessmentGroupsModalLabel">
+                                  Upload new group assignments
+                                </h1>
+                              </div>
+                              <div class="modal-body">
+                                <p>Upload a CSV file in the format of:</p>
+                              </div>
+                              <div class="modal-body">
+                                <code class="mb-0 text-dark">
+                                  groupName,UID<br />
+                                  groupA,one@example.com<br />
+                                  groupA,two@example.com<br />
+                                  groupB,three@example.com<br />
+                                  groupB,four@example.com</code
```

Comment:
Nit: make this a `pre`?

Reply:
Is there a reason to use `pre` here? it is going to preserve whitespace, which seems unnecessary.